gateway/src/apicast/policy/liquid_context_debug/context_content.lua

```diff
@@ -54,7 +61,7 @@ local value_from_fun = {
       local wanted_types = accepted_types_for_keys[type(k)] and
                            accepted_types_for_values[type(v)]
 
-      if wanted_types then
+      if wanted_types and k ~= liquid_template_context_key then
```

Comment:
What about hiding the whole TemplateString? 

Also, I think we will need some protection from recursive structures in general. It should be as easy as a table that gets passed to record all elements we've seen already.

Reply:
Hiding TemplateString instances is not so straightforward because every policy could give them a different name.

Reply:
We can do that by checking the metatable of the object.

Reply:
Regarding what you mentioned about avoid recursive structures in general, I agree. However, that alone would not solve the specific case of the liquid templates. It's a bit tricky. The context stored in the template string instance could be different from the context that we have in the liquid context debug policy. The context in the template string instance is stored when we call render() on the template. The context at that moment could be different from any other context received in other policies, including the context debug one, because any policy can modify the context in any of the phases.